contracts/BC_fusion/StBNB.sol

```diff
@@ -0,0 +1,354 @@
+// SPDX-License-Identifier: GPL-3.0-or-later
+pragma solidity ^0.8.0;
+
+import "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+import "@openzeppelin/contracts-upgradeable/utils/ContextUpgradeable.sol";
+
+/**
+ * @title Interest-bearing ERC20-like token for BSC New Staking protocol.
+ *
+ * This contract is abstract. To make the contract deployable override the
+ * `_getTotalPooledBNB` function. `StakePool.sol` contract inherits StBNB and defines
+ * the `_getTotalPooledBNB` function.
+ *
+ * StBNB balance only changes on transfers and represents the holder's share of the
+ * total staked funds. Account shares aren't normalized, so the
+ * contract also stores the sum of all shares. When staking happens, the user's shares
+ * are calculated as:
+ *
+ * shares[account] = stakedAmount * _getTotalShares() / _getTotalPooledEther()
+ *
+ * The token inherits from `PausableUpgradeable` and uses `whenNotStopped` modifier for methods
+ * which change `_shares` or `_allowances`. `_stop` and `_resume` functions are overridden
+ * in `StakePool.sol` and might be called by an account with the `PAUSE_ROLE`. This is useful
+ * for emergency scenarios, e.g. a protocol bug, where one might want
+ * to freeze all token transfers and approvals until the emergency is resolved.
+ */
+abstract contract StBNB is IERC20, ContextUpgradeable {
```

Comment:
Do we really need `ContextUpgradeable` here?

1. I think we upgrade contract through hardfork.
2. `ContextUpgradeable` has  `Initializable` interface, but we dont know who to do the initialize here?